examples/operator.rs

```diff
@@ -0,0 +1,100 @@
+extern crate timely;
+extern crate timely_communication;
+
+use std::collections::HashMap;
+
+use timely_communication::Configuration;
+use timely::dataflow::operators::{ToStream, Input, Inspect, Operator, FrontierNotificator};
+use timely::dataflow::channels::pact::Pipeline;
+use timely::progress::timestamp::RootTimestamp;
+use timely::dataflow::Scope;
+
+fn main() {
+    timely::execute(Configuration::Thread, |root| {
+        root.scoped(|scope| {
+            (0u64..10).to_stream(scope)
+                .unary_frontier(Pipeline, "example", |mut builder| {
+                    let mut cap = Some(builder.get_cap(RootTimestamp::new(12)));
+                    move |(input, _frontier), output| {
+                        cap = None;
+                        while let Some((time, data)) = input.next() {
+                            output.session(&time).give_content(data);
+                        }
+                    }
+                }).inspect(|x| println!("{:?}", x));
+        });
+    }).unwrap();
+
+    timely::execute(Configuration::Thread, |root| {
+        root.scoped(|scope| {
+            (0u64..10).to_stream(scope)
+                .unary(Pipeline, "example", |mut builder| {
```

Comment:
Thanks! Elbows deep in diff dataflow stuff at the moment, but will try and check out code today. Nested closures does sound better than alternative (and better explains actual ownership).

About 2.: Sorry, I should have looked harder. I assumed because it managed the frontiers it had to be blessed so that `push_external_progress` would know to have a chat with it and update the frontiers. Let me take a peek, but that was the only magical thing I thought needed to be removed.

About 3.: no clue. Sorry, that was a bit more speculative and maybe nonsense. I've often found myself with a big pile of `(data, time)` tuples, and want to track down the associated capabilities which I just keep in a list, and I was trying to think about some place sane to put them.